weaviate/collections/aggregations/aggregate.py

```diff
@@ -77,6 +88,82 @@ def _to_aggregate_result(
+            raise WeaviateQueryError("No results found in the aggregation query!", "gRPC")
+        if is_group_by:
+            AggregateGroupByReturn(
+                groups=[
+                    AggregateGroup(
+                        grouped_by=self.__parse_grouped_by_value(group.grouped_by),
+                        properties={
+                            aggregation.property: self.__parse_property_grpc(aggregation)
+                            for aggregation in group.aggregations.aggregations
+                        },
+                        total_count=group.objects_count,
+                    )
+                    for group in response.result.groups
+                ]
+            )
+        else:
+            result = response.result.groups[0]
+            return AggregateReturn(
+                properties={
+                    aggregation.property: self.__parse_property_grpc(aggregation)
+                    for aggregation in result.aggregations.aggregations
+                },
+                total_count=result.objects_count,
+            )
+        return AggregateGroupByReturn(
+            groups=[
+                AggregateGroup(
+                    grouped_by=self.__parse_grouped_by_value(group.grouped_by),
+                    properties={
+                        aggregation.property: self.__parse_property_grpc(aggregation)
+                        for aggregation in group.aggregations.aggregations
+                    },
+                    total_count=group.objects_count,
+                )
+                for group in response.result.groups
+            ]
+        )
+
+    def __parse_grouped_by_value(
+        self, grouped_by: aggregate_pb2.AggregateGroup.GroupedBy
+    ) -> GroupedBy:
+        value: Union[
+            str, int, float, bool, List[str], List[int], List[float], List[bool], GeoCoordinate
+        ]
+        if grouped_by.HasField("text"):
+            value = grouped_by.text
+        elif grouped_by.HasField("int"):
+            value = grouped_by.int
+        elif grouped_by.HasField("number"):
+            value = grouped_by.number
+        elif grouped_by.HasField("boolean"):
+            value = grouped_by.boolean
+        elif grouped_by.HasField("texts"):
+            value = list(grouped_by.texts.values)
+        elif grouped_by.HasField("ints"):
+            value = list(grouped_by.ints.values)
+        elif grouped_by.HasField("numbers"):
+            value = list(grouped_by.numbers.values)
+        elif grouped_by.HasField("booleans"):
+            value = list(grouped_by.booleans.values)
+        elif grouped_by.HasField("geo"):
+            v = grouped_by.geo
+            value = GeoCoordinate(
+                latitude=v.latitude,
+                longitude=v.longitude,
+            )
+        else:
+            raise ValueError(
+                f"Unknown grouped by type {grouped_by} encountered in _Aggregate.__parse_grouped_by_value()"
+            )
```

Comment:
not sure about this, this could craete problems when we add a new field, that older versions will throw an exception. We had that with get when we introduced an extra is_none type

Reply:
The alternative is to set `value` to a null, e.g. either `None` or `""`, to therefore implicitly fail rather than explicitly fail. Which do you think is preferable from a UX PoV?

Reply:
what did we do with .query?

Reply:
Ah, we do this:
```python
if value.HasField("null_value"):
    return None

_Warnings.unknown_type_encountered(value.WhichOneof("Value"))
return None
```
because `None` is an acceptable value there. I had it that the `grouped_by.value` cannot be `None` of the type system but that may be too restrictive. I'll use this pattern in aggregate

Reply:
I think it's fine since that type is already a `Union` of all the other types so users will have to type coerce anyway